artemis-cli/src/test/java/org/apache/activemq/artemis/cli/commands/CreateTest.java

```diff
@@ -0,0 +1,105 @@
+ */
+package org.apache.activemq.artemis.cli.commands;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.activemq.artemis.utils.XmlProvider;
+import org.apache.activemq.cli.test.CliTestBase;
+import org.apache.activemq.cli.test.TestActionContext;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.xml.sax.SAXException;
+
+@RunWith(Parameterized.class)
+public class CreateTest extends CliTestBase {
+
+   private final String testName;
+   private String httpHost;
+   private boolean relaxJolokia;
+
+   public CreateTest(String testName, String httpHost, boolean relaxJolokia) throws IOException {
+      this.testName = testName;
+      this.httpHost = httpHost;
+      this.relaxJolokia = relaxJolokia;
+   }
+
+   @Parameterized.Parameters(name = "{0}")
+   public static Collection<Object[]> testData() {
+      return Arrays.asList(new Object[][]{
+         {"Happy path + relaxJolokia", "sampledomain.com", true},
+         {"Happy path - relaxJolokia", "sampledomain.net", false},
+         {"Domain with dash + relaxJolokia", "sample-domain.co", true},
+         {"Domain with dash - relaxJolokia", "sample-domain.co.uk", false},
+         {"Domain with double dashes + relaxJolokia", "sample--domain.name", true},
+         {"Domain with double dashes - relaxJolokia", "sample--domain.biz", false},
+         {"Domain with leading dashes + relaxJolokia", "--sampledomain.company", true},
+         {"Domain with leading dashes - relaxJolokia", "--sampledomain.email", false},
+         {"Domain with trailing dashes + relaxJolokia", "sampledomain--.shop", true},
+         {"Domain with trailing dashes - relaxJolokia", "sampledomain--.java", false},
+      });
+   }
+
+   @Test
+   public void testWriteJolokiaAccessXmlCreatesValidXml() throws Exception {
+      TestActionContext context = new TestActionContext();
+      File testInstance = new File(temporaryFolder.getRoot(), "test-instance");
+
+      Create c = new Create();
+      c.setNoAutoTune(true);
+      c.setInstance(testInstance);
+      c.setHttpHost(httpHost);
+      c.setRelaxJolokia(relaxJolokia);
+      c.execute(context);
+
+      try {
+         isXmlValid(new File(testInstance, "etc/" + Create.ETC_JOLOKIA_ACCESS_XML));
+      } catch (Exception e) {
+         Assert.fail(testName + " should be valid, but " + e.toString());
+      }
+   }
+
+   /**
+    * IsXmlValid will check if a given xml file is valid by parsing the xml to create a Document.
+    * <p>
+    * If it parses, the xml is assumed to be valid. If any exceptions occur, the xml is not valid.
+    *
+    * @param xml The xml file to check for validity.
+    * @return whether the xml file represents a valid xml document.
+    */
+   private boolean isXmlValid(File xml) {
```

Comment:
Note: I left this as a separate method to make it more explicit what was being done.